### PR TITLE
[DCOS-38237] Use disambiguation suffixes for drivers and executors.

### PR DESCRIPTION
**Pull request by porridge**
Unfortunately we still need both flags, as `--appName` affects the
drivers, while `--name` affects the executors.

Example:
```
$ dcos task
NAME                                            HOST            USER   STATE  ID                                         MESOS ID                                  REGION  ZONE
Driver for KafkaRandomFeeder-k00-p00            172.31.16.235   root     R    driver-20180619132306-0025                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S8    ---    ---
Driver for KafkaRandomFeeder-k00-p01            172.31.21.197   root     R    driver-20180619132314-0027                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S9    ---    ---
Driver for KafkaRandomFeeder-k00-p02            172.31.20.117   root     R    driver-20180619132321-0029                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S6    ---    ---
Driver for KafkaRandomFeeder-k00-p03            172.31.38.220   root     R    driver-20180619132329-0031                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S7    ---    ---
Driver for KafkaRandomFeeder-k01-p00            172.31.40.142   root     R    driver-20180619132338-0025                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S5    ---    ---
Driver for KafkaRandomFeeder-k01-p01            172.31.38.220   root     R    driver-20180619132346-0027                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S7    ---    ---
Driver for KafkaRandomFeeder-k01-p02            172.31.15.226   root     R    driver-20180619132353-0029                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S2    ---    ---
Driver for KafkaRandomFeeder-k01-p03            172.31.21.197   root     R    driver-20180619132400-0031                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S9    ---    ---
Driver for KafkaWordCount-k00-p00-c00           172.31.20.117   root     R    driver-20180619132310-0026                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S6    ---    ---
Driver for KafkaWordCount-k00-p01-c00           172.31.16.235   root     R    driver-20180619132317-0028                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S8    ---    ---
Driver for KafkaWordCount-k00-p02-c00           172.31.15.226   root     R    driver-20180619132325-0030                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S2    ---    ---
Driver for KafkaWordCount-k00-p03-c00           172.31.21.197   root     R    driver-20180619132333-0032                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S9    ---    ---
Driver for KafkaWordCount-k01-p00-c00           172.31.21.197   root     R    driver-20180619132342-0026                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S9    ---    ---
Driver for KafkaWordCount-k01-p01-c00           172.31.40.142   root     R    driver-20180619132350-0028                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S5    ---    ---
Driver for KafkaWordCount-k01-p02-c00           172.31.40.142   root     R    driver-20180619132357-0030                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S5    ---    ---
Driver for KafkaWordCount-k01-p03-c00           172.31.16.235   root     R    driver-20180619132405-0032                 7d368349-375a-4fd0-9b25-0034ecb53ad7-S8    ---    ---
KafkaRandomFeeder-k00-p00 0                     172.31.38.220   root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S7    ---    ---
KafkaRandomFeeder-k00-p01 0                     172.31.32.170   root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S1    ---    ---
KafkaRandomFeeder-k00-p02 0                     172.31.21.197   root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S9    ---    ---
KafkaRandomFeeder-k00-p03 0                     172.31.21.197   root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S9    ---    ---
KafkaRandomFeeder-k01-p00 0                     172.31.15.226   root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S2    ---    ---
KafkaRandomFeeder-k01-p01 0                     172.31.20.117   root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S6    ---    ---
KafkaRandomFeeder-k01-p02 0                     172.31.32.170   root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S1    ---    ---
KafkaRandomFeeder-k01-p03 0                     172.31.40.142   root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S5    ---    ---
KafkaWordCount-k00-p00-c00 0                    172.31.38.220   root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S7    ---    ---
KafkaWordCount-k00-p01-c00 0                    172.31.8.75     root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S3    ---    ---
KafkaWordCount-k00-p02-c00 0                    172.31.15.226   root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S2    ---    ---
KafkaWordCount-k00-p03-c00 0                    172.31.8.75     root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S3    ---    ---
KafkaWordCount-k01-p00-c00 0                    172.31.15.226   root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S2    ---    ---
KafkaWordCount-k01-p01-c00 0                    172.31.16.235   root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S8    ---    ---
KafkaWordCount-k01-p02-c00 0                    172.31.15.242   root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S0    ---    ---
KafkaWordCount-k01-p03-c00 0                    172.31.32.170   root     R    0                                          7d368349-375a-4fd0-9b25-0034ecb53ad7-S1    ---    ---
[...]
```

### CODE REVIEW

**Comment by mpereira** [comment]
Would it be useful from a metrics/monitoring perspective to have `-driver` and `-executor` suffixes too?

**Comment by porridge** [comment]
I don't think these actually make it as far as the metrics labels.

File: scale-tests/kafka_cassandra_streaming_test.py [hunk 1]
```diff
@@ -95,6 +96,7 @@ def _submit_producer(jar,
 
     app_config = ["--conf",  "spark.cores.max={}".format(spark_cores_max),
                   "--conf",  "spark.executor.cores={}".format(spark_executor_cores),
+                  "--name",  name,
```
Comment (elezar) [inline scale-tests/kafka_cassandra_streaming_test.py]: Were both of these required?

**Comment by porridge** [comment]
Merging now, I believe @elezar does not object.

**Pull request merged**